Standard.AI.Data.EntityIntelligence.Tests.Unit/Services/Foundations/Datas/DataServiceTests.Exceptions.Retrieve.cs

```diff
@@ -0,0 +1,148 @@
+                expectedDataDependencyValidationException);
+
+            this.dataBrokerMock.Verify(broker =>
+                broker.ExecuteQueryAsync<TableColumnMetadata>(It.IsAny<string>()),
+                    Times.Once);
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyValidationExceptionOnRetrieveIfInvalidOperationExceptionOccursAsync()
+        {
+            // given
+            var invalidOperationException = new InvalidOperationException();
+
+            var invalidOperationDataValidationException =
+                new InvalidOperationDataValidationException(invalidOperationException);
+
+            var expectedDataDependencyValidationException =
+                new DataDependencyValidationException(
+                        invalidOperationDataValidationException);
+
+            this.dataBrokerMock.Setup(broker =>
+                broker.ExecuteQueryAsync<TableColumnMetadata>(It.IsAny<string>()))
+                    .ThrowsAsync(invalidOperationException);
+
+            // when
+            var retrieveTablesDetailsTask = this.dataService.RetrieveTablesDetailsAsync();
+
+            var actualTableDetailsException =
+                await Assert.ThrowsAsync<DataDependencyValidationException>(
+                                       retrieveTablesDetailsTask.AsTask);
+
+            // then
+            actualTableDetailsException.Should().BeEquivalentTo(
+                expectedDataDependencyValidationException);
+
+            this.dataBrokerMock.Verify(broker =>
+                broker.ExecuteQueryAsync<TableColumnMetadata>(It.IsAny<string>()),
+                    Times.Once);
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnSqlException()
+        {
+            // given
+            SqlException sqlException = GetSqlException();
+
+            var failedDataDependencyException =
+                new FailedDataDependencyValidationException(sqlException);
+
+            var expectedDataDependencyException =
+                new DataDependencyException(
+                    failedDataDependencyException);
+
+            this.dataBrokerMock.Setup(broker =>
+                broker.ExecuteQueryAsync<TableColumnMetadata>(It.IsAny<string>()))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            var retrieveTablesDetailsTask = this.dataService.RetrieveTablesDetailsAsync();
+
+            var actualTableInformationListException =
+                await Assert.ThrowsAsync<DataDependencyException>(
+                                       retrieveTablesDetailsTask.AsTask);
+
+            // then
+            actualTableInformationListException.Should().BeEquivalentTo(
+                expectedDataDependencyException);
+
+            this.dataBrokerMock.Verify(broker =>
+                broker.ExecuteQueryAsync<TableColumnMetadata>(It.IsAny<string>()),
+                    Times.Once);
+        }
```

Comment:
Verify no other calls on the broker.
```suggestion
            this.dataBrokeMock.VerifyNoOtherCalls();
        }
```